### PR TITLE
Implement Dropout

### PR DESCRIPTION
**Pull request by MatthieuHernandez**
The Dropout seems well implemented but I'm not able to reproduce the result of this paper :
 * [Dropout: A Simple Way to Prevent Neural Networks from Overfitting](https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf)

As long as the Dropout doesn't improve any datasets test, no merge this pull request.

### CODE REVIEW

**Comment by MatthieuHernandez** [comment]
⚠️ Don't forget to revert the MNIST tests changes

**Comment by MatthieuHernandez** [comment]
I will merge this pull request even if no test from this project proves the effectiveness of Dropout.
